types/part_set_test.go

```diff
@@ -114,9 +114,17 @@ func TestEncodingDecodingRoundTrip(t *testing.T) {
 			eps, lastPartLen, err := Encode(ops, BlockPartSizeBytes)
 			require.NoError(t, err)
 
-			ops.parts[0] = nil
+			// Remove part 0 to test decode functionality
+			ops.mtx.Lock()
 			ops.count--
 			ops.partsBitArray.SetIndex(0, false)
+			// Clear the buffer area for part 0
+			start := 0 * int(BlockPartSizeBytes)
```

Comment:
does it make sense to have a method to remove a part from the partset? something like what: `ops.parts[0] = nil` would achieve?

Reply:
It's used only in this single test, I would leave it as is.